filters/src/main/java/ua/net/tokar/json/rainbowrest/JsonResponseWrapper.java

```diff
@@ -9,13 +9,20 @@
 import java.io.IOException;
 import java.io.OutputStreamWriter;
 import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
 
-class HtmlResponseWrapper extends HttpServletResponseWrapper {
+class JsonResponseWrapper extends HttpServletResponseWrapper {
```

Comment:
I think it should be a HttpResponseWrapper

Reply:
rainbow doesn’t work with just any http responses. plus, we have an assumption about the content being json, so I don’t agree

Reply:
there is no json specifics except your comment in the class :)

Reply:
Also, the only reference to json is at https://github.com/alexeytokar/rainbow-rest/pull/39/files/02d978a97d6d29bbed22822fd1b0f9527e7a6074#diff-98a7774d20b34025ac7ef3669f219e89R125 and probably that guess should be moved there as well

Reply:
there are
1. we assume response charset being UTF-8 when we ignore default ISO-8859-1
2. we unconditionally [parse content as JSON](https://github.com/alexeytokar/rainbow-rest/pull/39/files#diff-6791de1d7311ef2e990db0aee36a441aR151)

Reply:
(the first breaks the Servlet spec, afaik)

Reply:
I think it would be ok to name it just `HttpResponseWrapper` if we move decoding out of it and return raw bytes, like [ContentCachingResponseWrapper](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/util/ContentCachingResponseWrapper.html) does.

Reply:
...or we can reuse charEncoding from the initial response that is decorated

Reply:
what do you mean?

Reply:
I mean that constructer can be adjusted to set charset from ServletResponse response.getCharacterEncoding() for example

Reply:
I’m sorry, I don’t see how that would work.

calling `response.getCharacterEncoding()` in the constructor of a wrapper will always return default ISO-8859-1, because it’s too early, the request handler hadn’t have a chance to set it.

Reply:
oh. but I still do not get why the server which creates a response does not set proper encoding and why it becomes a problem of Rainbow?

Reply:
I see. Skip prev comment